modules/core/src/lib/utils/response-utils.ts

```diff
@@ -77,20 +77,24 @@ export function checkResponseSync(response: Response): void {
 
 // HELPERS
 
-async function getResponseError(response: Response): Promise<string> {
-  let message = `Failed to fetch resource ${response.url} (${response.status}): `;
+async function getResponseError(response: Response): Promise<Error> {
+  const message = `${response.status} ${response.statusText}: Failed to fetch resource`;
+  const cause: {
+    url: string;
+    response?: any;
+  } = {url: response.url};
   try {
     const contentType = response.headers.get('Content-Type');
-    let text = response.statusText;
     if (contentType?.includes('application/json')) {
-      text += ` ${await response.text()}`;
+      cause.response = await response.json();
+    } else {
+      cause.response = await response.text();
     }
```

Comment:
json can be very messy and if this is e.g. xml it can get crazy. This is part of why including this verbatim is so problematic.